test/phoenix_live_view/router_test.exs

```diff
@@ -60,4 +60,20 @@ defmodule Phoenix.LiveView.RouterTest do
     assert Routes.foo_bar_nested_index_path(conn, :show) == "/router/foobarbaz/nested/show"
     assert Routes.custom_foo_bar_path(conn, :index) == "/router/foobarbaz/custom"
   end
+
+  test "user-defined metadata is available inside of metadata key" do
+    assert Phoenix.Router.route_info(Phoenix.LiveViewTest.Router, "GET", "/opts-with-metadata", nil) == %{
+      log: :debug,
+      path_params: %{},
+      phoenix_live_view: {Phoenix.LiveViewTest.OptsLive, nil},
+      pipe_through: [],
+      plug: Phoenix.LiveView.Plug,
+      plug_opts: Phoenix.LiveViewTest.OptsLive,
+      route: "/opts-with-metadata",
+      route_name: "opts"
```

Comment:
Oh, let's note compare all keys, because if Phoenix adds new keys, then it will break. Let's just assert for the key we have added ourselves (in this case, "route_name").

Reply:
Thanks, I got it; It's good to know this kind of practices.